08-software-versions.Rmd

```diff
@@ -5,6 +5,108 @@
+```
+
+As developers and maintainers of software continue to make updates and fixes to the software, the developers and maintainers of other interdependent software are doing similarly, meaning that software dependencies and the computing environments are not only a complicated mess at times, but also a moving target!
+
+## Software and package versions affect results!
+
+Sometimes if we have generally the same software installed for reproducing an analysis, we may feel that that is "close enough". And given all the other technical aspects of reproducibility, it can be easy to overlook what versions of software packages we are using. However, controlling for software versions is critical for creating reproducible analyses. Software versions can directly affect not only whether an analysis will be able to run, but the results of the analysis [@BeaulieuJones2017].
+
+## Session Info
+
+Perhaps the easiest way to begin to address computing environment variability is to record what the computing environment looks like at the time an analysis is run. In R, this is a fairly straightforward task.
+
+Generally at the end of your R notebook, you will want to print out your session info. You can do this by running the function `sessionInfo()` or the tidyverse version of this function from the devtools package, `devtools::session_info()`.
+
+We can run sessionInfo in this book (this book is powered by R).
+
+```{r}
+sessionInfo()
+```
+
+Now we have recorded what some key aspects of our computing environment looked like at the time that this book was rendered last.
+This print out may seem like a lot of nonsense at first, but it gives us some useful information in a pinch!
+
+
+```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "Reproducible parrot is confused because their results are different after they have re-run their analysis. The parrot says: ‘Hmm… why am I getting a different result this time? Good thing I can check the session info to see if package versions might have caused this change!’"}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g201bd406763_37_1030")
+```
+
+If we take a look at two different session info print outs, we can begin to spot the differences. These differences may give us clues into why an analysis ran differently.
+
+```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "Two session info print outs are show side by side.Highlighted we can see that they have different R versions: 4.0.2 vs 4.0.5. They also have different operating systems. The packages they have attached is rmarkdown but they also have different rmarkdown package versions!  If there are  discrepancies in re-runs of the analysis, the session info print out gives a record which may have clues to why that might be! This can give items to look into for determining why the results didn’t reproduce as expected."}
```

Comment:
```suggestion
```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "Two sessionInfo printouts are show side by side.Highlighted we can see that they have different R versions: 4.0.2 vs 4.0.5. They also have different operating systems. The packages they have attached is rmarkdown but they also have different rmarkdown package versions!  If there are  discrepancies in re-runs of the analysis, the sessionInfo printout gives a record which may have clues to why that might be! This can give items to look into for determining why the results didn’t reproduce as expected."}
```

Reply:
I will edit away my sessionInfo edits